pwnlib/asm.py

```diff
@@ -267,8 +268,8 @@ def _assembler():
         'ia64':    [gas, '-m%ce' % context.endianness[0]],
 
         # riscv64-unknown-elf-as supports riscv32 as well as riscv64
-        'riscv32': [gas, '-march=rv32gc', '-mabi=ilp32'],
-        'riscv64': [gas, '-march=rv64gc', '-mabi=lp64'],
+        'riscv32': [gas, '-march=rv32g', '-mabi=ilp32'],
```

Comment:
This prevents you from using compressed instructions when manually writing shellcode and compiling using `asm`, no?

Reply:
Sure, and therefore I am not very sure about this change. I don't know any way to avoid C and yet allow c.* opcodes.

Reply:
I think we thought about adding assembler command line args to context when adding riscv support? Or just `asm`?

Reply:
We could do this like we do with thumb (separate arch) but I doubt whether it is worth it. Maybe we can revisit this one day.

Reply:
Would it make sense to make two different archs? That way you could do `asm -c riscv32` and `asm -c riscv32c` or something similar?

Reply:
Maybe add this as a context.subarch similar to what is requested for mips and arm? It's not really a sub-architecture though